packages/netlify-cms-backend-github/src/API.ts

```diff
@@ -285,108 +421,112 @@ export default class API {
 
       if (!this.useOpenAuthoring) {
         return this.request(`${this.repoURL}/contents/${key}.json`, metadataRequestOptions)
-          .then(response => JSON.parse(response))
+          .then((response: string) => JSON.parse(response))
           .catch(errorHandler);
       }
 
       const [user, repo] = key.split('/');
       return this.request(`/repos/${user}/${repo}/contents/${key}.json`, metadataRequestOptions)
-        .then(response => JSON.parse(response))
+        .then((response: string) => JSON.parse(response))
         .catch(errorHandler);
     });
   }
 
-  retrieveContent(path, branch, repoURL) {
+  retrieveContent({ path, branch, repoURL, parseText }: ContentArgs) {
```

Comment:
`retrieveContent`will now always get the blob `sha` first, then use that to get the blob.
This way the blob will also be cached locally.